Octokit/Clients/Enterprise/EnterpriseProbe.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Service used to check if a GitHub Enterprise instance lives at a given URL.
+    /// </summary>
+    public class EnterpriseProbe : IEnterpriseProbe
+    {
+        static readonly Uri endPoint = new Uri("/site/sha", UriKind.Relative);
+        readonly ProductHeaderValue productHeader;
+        readonly IHttpClient httpClient;
+
+        public EnterpriseProbe(ProductHeaderValue productHeader, IHttpClient httpClient)
```

Comment:
Can we add an overload that doesn't take the `httpClient` and instead defaults it to `new HttpClientAdapter(HttpMessageHandlerFactory.CreateDefault)` 

This is consistent with the way most normal users would be using octokit, e.g they dont really have any idea about `IHttpClient` or the fact that a default one can be created using the `HttpMessageHandlerFactory`.  And yes I do acknowledge the documentation example does show how to use the handler factory, but I think it would be more inline with the overloads on `GitHubClient` (well, really it's the `Connection` class) to not have to specify it for regular use